lib/app/pull.go

```diff
@@ -266,3 +288,43 @@ func (r Puller) pullApp(loc loc.Locator) error {
 	}
 	return trace.Wrap(err)
 }
+
+func onConflictDependencies(upsert bool) conflictHandler {
+	if upsert {
+		return onConflictContinue
+	}
+	return onConflictSkip
+}
+
+func onConflict(upsert bool) conflictHandler {
```

Comment:
Nit: I would add comments on this and the above functions explaining our reasoning behind handling "already exists" cases for the app and its dependencies.